### PR TITLE
Accept keyhashes for required signers

### PR DESCRIPTION
**Pull request by Jimbo4350**
Required signers can now be specified via the `--required-signer-hash` cli option
Resolves: #3338

### CODE REVIEW

**Comment by mkoura** [comment]
QA tested this with the following plutus script with both normal and extended keys, and with key files and key hashes. Works as expected :+1: 
```haskell
mkPolicy :: PubKeyHash -> ScriptContext -> Bool
mkPolicy pkh ctx = traceIfFalse "not signed by redeemer pubkeyhash" checkWitness
  where
    info :: TxInfo
    info  = scriptContextTxInfo ctx

    checkWitness :: Bool
    checkWitness  = txSignedBy info pkh
```

File: cardano-cli/src/Cardano/CLI/Shelley/Parsers.hs [hunk 1]
```diff
@@ -1430,20 +1430,36 @@ pColdSigningKeyFile =
...
+      (  Opt.long "required-signer-hash"
+      <> Opt.metavar "HASH"
+      <> Opt.help "Hash of the verification key (zero or more) whose \
+                  \signature is required."
+      )
+  sExtendedPayKeyHash :: Parser (Hash PaymentExtendedKey)
+  sExtendedPayKeyHash =
+    Opt.option (readerFromParsecParser $ parseHash (AsHash AsPaymentExtendedKey))
+      (  Opt.long "required-signer-hash"
+      <> Opt.hidden
```
Comment (newhoggy) [inline cardano-cli/src/Cardano/CLI/Shelley/Parsers.hs:1460]: What is this option for?

Comment (Jimbo4350) [reply cardano-cli/src/Cardano/CLI/Shelley/Parsers.hs:1460]: Ah I actually don't need this

**Review by newhoggy — APPROVED** [review]

**Comment by Jimbo4350** [comment]
bors merge

**Pull request merged**